src/core/mac/mac_header_ie.hpp

```diff
@@ -275,18 +342,72 @@ class TimeIe : public VendorIeHeader
     void SetTime(uint64_t aTime) { mTime = LittleEndian::HostSwap64(aTime); }
 
 private:
-    uint8_t  mSequence;
-    uint64_t mTime;
+    NestIeHeader mNestIeHeader;
```

Comment:
It seems like you changed the model so that the specialized IE class types no longer inherit from `HeaderIe` (or other base classes) and instead include it as part of their fields.

While this works, is there any benefit to this approach? It seems natural to me that `TimeIe` or `EnhAckProbingIe` would inherit from the `HeaderIe` base class and provide a common set of methods (like `GetId()`, `GetLength()`, etc.). We use this model for TLVs.